src/planscape/planning/views_v2.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from rest_framework import viewsets
+
+from planning.models import PlanningArea, Scenario
+from planning.serializers import (
+    PlanningAreaSerializer,
+    ListPlanningAreaSerializer,
+    ListScenarioSerializer,
+    ScenarioSerializer,
+)
+from planning.filters import PlanningAreaFilter, ScenarioFilter
+from planning.permission import PlanningAreaViewPermission, ScenarioViewPermission
+
+logger = logging.getLogger(__name__)
+
+
+class PlanningAreaViewSet(viewsets.ModelViewSet):
+    queryset = PlanningArea.objects.all()
+    permission_classes = [PlanningAreaViewPermission]
+    ordering_fields = ["name", "created_at", "scenario_count"]
+    filterset_class = PlanningAreaFilter
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return ListPlanningAreaSerializer
+        return PlanningAreaSerializer
+
+    def get_queryset(self):
+        user = self.request.user
+        qs = PlanningArea.objects.get_for_user(user)
+        return qs
+
+
+class ScenarioViewSet(viewsets.ModelViewSet):
+    queryset = Scenario.objects.all()
+    permission_classes = [ScenarioViewPermission]
+    ordering_fields = ["name", "created_at"]
+    filterset_class = ScenarioFilter
+
+    def create(self, request, planningarea_pk):
+        request_data = request.data
+        request_data["planning_area"] = planningarea_pk
```

Comment:
overriding `create` seemed to be the only viable approach for setting the `planning_area` value from the `planningarea_pk` path param, since apparently the serializer validation is called before `perform_create`. Does this seem right?